### PR TITLE
Support Sidekiq 7

### LINKED ISSUE #10: Support Sidekiq 7.x
Sidekiq 7.x removed 6.x's `lib/sidekiq/worker.rb` file, leading to this kind of error:

```
$ rails test
/usr/local/bundle/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:17:in `require': cannot load such file -- sidekiq/worker (LoadError)
        from /usr/local/bundle/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:17:in `require'
        from /usr/local/bundle/gems/sidekiq-staged_push-0.1.0/lib/sidekiq/staged_push.rb:4:in `<main>'
[...]
```

Simply changing `lib/sidekiq/staged_push` to
```
require "sidekiq/worker_compatibility_alias"
```

lets my app tests/examples run, but maybe there's more you'd want to check/do? 🤷

### PR DESCRIPTION
**Pull request by bobmazanec**
Intent is to fix #10

Mostly "housekeeping" basics so far
I've looked at the [blob](https://github.com/mperham/sidekiq/blob/ad0f13cac5138b891d54ef2f2df3460d5151e34c/lib/sidekiq/transaction_aware_client.rb#L41) you mentioned in the bug discussion, but don't get the vision/direction yet.

Great [Development](https://github.com/adamniedzielski/sidekiq-staged_push#development) docs & setup—dev setup was a snap.
I'm quite new to Sidekiq internals and extensions.
If you're willing, maybe describe the spec/example changes you're looking for—🤞 that'll help me see how things fit and what will need to change.

Thanks

### CODE REVIEW

**Comment by bobmazanec** [comment]
I forgot to mentioned that I added `"<7"` to the gem spec to make sure I got it working against Sidekiq [6.5, 7.0) before pressing on to 7.x by removing that.

File: sidekiq-staged_push.gemspec [hunk 1]
```diff
@@ -30,7 +30,7 @@ Gem::Specification.new do |spec|
   spec.required_ruby_version = ">= 2.7.5"
 
   spec.add_dependency "activerecord", ">= 5.2.0"
-  spec.add_dependency "sidekiq", ">= 6.3.1"
+  spec.add_dependency "sidekiq", ">= 6.5.0", "<7"
```
Comment (adamniedzielski) [inline sidekiq-staged_push.gemspec:33]: There's no need to bump the minimum version to 6.5.0 for the particular change that you made. `sidekiq/job` should already be available on 6.3.1

**Review by adamniedzielski — COMMENTED** [review]
Thanks for contributing @bobmazanec 💛 And thank you for your kind words about the dev setup.

What you did should already fix #10. The tests are failing with some cryptic message when running `bundle install`, though. I think that you missed to commit the change with bumping the version, but I'm not sure if it's enough to resolve the errors. I may have to look into that.

What I meant in https://github.com/adamniedzielski/sidekiq-staged_push/issues/10#issuecomment-1385945305 is to replace:

```ruby
Sidekiq::Worker::ClassMethods.prepend Sidekiq::StagedPush::Worker::ClassMethods
```

with:

```ruby
default_job_options["client_class"] = Sidekiq::StagedPush::Client
Sidekiq::JobUtil::TRANSIENT_ATTRIBUTES << "client_class"
```

It requires some further changes, though, because the interface of `Sidekiq::StagedPush::Client` needs to become compatible with `Sidekiq::Client`.

Feel free to experiment with that if you want, but the change that you submitted should already fix the immediate problem 🙂

File: sidekiq-staged_push.gemspec (hunk 1 above)
Comment (bobmazanec) [reply sidekiq-staged_push.gemspec:33]: Thanks for the info & explanation.
I thought about bumping the gem version, but didn't know if you'd want other stuff done/merged before prepping a release 🤷‍♂️ 

I'll revert the dependency version and play around to confirm(/refute?) this change is enough to fix #10. Either way I expect it'll be educational. 🤓 

Sounds like I can (should?) step away from the "additional changes" in this PR and let you "manage" them as you see fit—e.g., personal TODO, new/separate issue, ignore, ...

Thanks!

**Comment by bobmazanec** [comment]
[🤦‍♂️ all this didn't really belong in my response to your inline comment]
Thanks for the info & explanation.
I thought about bumping the gem version, but didn't know if you'd want other stuff done/merged before prepping a release 🤷‍♂️ 

I'll revert the dependency version and play around to confirm(/refute?) this change is enough to fix #10. Either way I expect it'll be educational. 🤓 

Sounds like I can (should?) step away from the "additional changes" in this PR and let you "manage" them as you see fit—e.g., personal TODO, new/separate issue, ignore, ...

Thanks!

**Comment by adamniedzielski** [comment]
> I thought about bumping the gem version, but didn't know if you'd want other stuff done/merged before prepping a release

I can do that. You just need to revert your changes to `Gemfile.lock`

> Sounds like I can (should?) step away from the "additional changes" in this PR and let you "manage" them as you see fit

Yes, I can do that later.

**Comment by adamniedzielski** [comment]
I did some improvements to the CI in https://github.com/adamniedzielski/sidekiq-staged_push/pull/12. It should be easier to make your changes now 🙂

**Comment by bobmazanec** [comment]
Reverted the `Gemfile.lock` & `.gemspec` changes, leaving only the `require` change.

**Comment by bobmazanec** [comment]
Rebased onto `main` cd7e417 and removed "<7" Sidekiq requirement

**Comment by bobmazanec** [comment]
Confirmed that specs pass under sidekiq 6.3.1, 6.5.8, and 7.0.3

**Comment by bobmazanec** [comment]
BTW, I left `sidekiq` @ 6.5.8 in `Gemfile.lock`
Can up it to 7.0.3 in this PR, if you like

**Comment by adamniedzielski** [comment]
Thanks!

> Can up it to 7.0.3 in this PR, if you like

Yes, please 🙏

**Comment by bobmazanec** [comment]
FWIW, `bundle outdated` now shows ("only"?)
```
Gem       Current  Latest   Requested  Groups
nokogiri  1.14.0   1.14.1
rack      2.2.6.2  3.0.4.1
tzinfo    2.0.5    2.0.6
```

**Review by adamniedzielski — APPROVED** [review]
Thank you! 💛

**Pull request merged**